playout/libretime_playout/timeout.py

```diff
@@ -14,27 +12,15 @@ def __init__(self):
         def run(self):
             self.result = func(*args, **kwargs)
 
-    first_attempt = True
-
     while True:
         thread = InterruptableThread()
         thread.start()
-        if not first_attempt:
-            timeout_duration = timeout_duration * 2
         thread.join(timeout_duration)
 
         if thread.is_alive():
-            # Restart Liquidsoap and try the command one more time. If it
-            # fails again then there is something critically wrong...
-            if first_attempt:
-                # restart liquidsoap
-                fetch.PypoFetch.ref.restart_liquidsoap()
-            else:
-                raise Exception("Thread did not terminate")
-        else:
-            return thread.result
-
-        first_attempt = False
+            raise Exception("Thread did not terminate")
+
+        return thread.result
```

Comment:
This whole timeout decoration might have become overkill, could probably be handled with a context manager and socket timeout.

Reply:
Will be refactored on a future PR